x-pack/plugins/ml/public/embeddables/single_metric_viewer/single_metric_viewer_controls_initializer.ts

```diff
@@ -39,17 +45,23 @@ export const initializeSingleMetricViewerControls = (
     titlesApi.setPanelTitle(update.panelTitle);
   };
 
-  const serializeSingleMetricViewerState = (): AnomalySwimLaneControlsState => {
+  const updateForecastId = (id: string | undefined) => {
+    forecastId.next(id);
+  };
+
+  const serializeSingleMetricViewerState = (): SingleMetricViewerControlsState => {
     return {
       jobIds: jobIds.value,
+      forecastId: forecastId.value,
```

Comment:
we should be using `getValue()` rather than accessing `value` directly for all of these items

Reply:
Good catch - updated in 53b573ff6d8d57acfe1ca468df74f5a8d63354a3